cloudsync/api.py

```diff
@@ -129,48 +171,94 @@ def get_error_type_from_et_error(et_error):
     return VideoStatus.TRANSCODE_FAILED_INTERNAL
 
 
-def refresh_status(video, encode_job=None):
+def refresh_status(video: Video, encode_job=None):
     """
     Check the encode job status & if not complete, update the status via a query to AWS.
-
     Args:
         video(ui.models.Video): Video object to refresh status of.
         encode_job(dj_elastictranscoder.models.EncodeJob): EncodeJob associated with Video
     """
     if video.status in (VideoStatus.TRANSCODING, VideoStatus.RETRANSCODING):
         if not encode_job:
             encode_job = video.encode_jobs.latest("created_at")
-        et_job = get_et_job(encode_job.id)
-        if et_job["Status"] == VideoStatus.COMPLETE:
-            process_transcode_results(video, et_job)
+        mc_job = get_media_convert_job(encode_job.id)
+        if mc_job["Job"]["Status"].lower() == VideoStatus.COMPLETE.lower():
+            with open("./config/results.json") as f:
+                results = prepare_results(video, encode_job, f.read())
+            process_transcode_results(results)
             video.update_status(VideoStatus.COMPLETE)
```

Comment:
We are updating the Video status in `process_transcode_results ` saving it again causing `post_save` signal to be triggered again and causing us to post same video to edX